### PR TITLE
[MRG+1] Harvard oxford

### LINKED ISSUE #914: fetch_atlas_harvard_oxford and symmetric_split
I have 2 little issues with `symmetric_split`:
1) the type of the output changes if `symmetric_split` is True or False
```Python
from nilearn import datasets
atlas = datasets.fetch_atlas_harvard_oxford('cort-maxprob-thr25-1mm', symmetric_split=True)
type(atlas['labels']), type(atlas['maps'])
```
(list, nibabel.nifti1.Nifti1Image)
```Python
atlas = datasets.fetch_atlas_harvard_oxford('cort-maxprob-thr25-1mm', symmetric_split=False)
type(atlas['labels']), type(atlas['maps'])
```
(numpy.ndarray, str)
2) the parameter `symmetric_split` is allowed to be True for `'sub-maxprob-thr25-1mm'` but 
```Python
datasets.fetch_atlas_harvard_oxford('sub-maxprob-thr25-1mm', symmetric_split=True)
```
throws an error `RuntimeError: data type not supported`.  May be not allowing for subcortical atlases as their labels already contain the information right/left

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
Fix #914 

There are still problems since some lateralized regions are splitted because they cross the inter-hemispheric limit but this PR, as for now, addresses most problems.

### CODE REVIEW

**Comment by lesteve** [comment]
Can we have tests for these fixes?

**Comment by AlexandreAbraham** [comment]
Yes we can. But as the tests shows, this is still WIP for the moment.

**Comment by lesteve** [comment]
> Yes we can. But as the tests shows, this is still WIP for the moment.

TDD ;-) ?

**Comment by AlexandreAbraham** [comment]
> TDD ;-) ?

Actually, my test is the real atlas. But I can't use it in the test suite so I've just put a test using toy data.

The PR should is now mergeable.

**Comment by GaelVaroquaux** [comment]
LGTM.

@salma1601 : do you want to check that it solves your problem?

**Comment by salma1601** [comment]
from what I checked, yes. Thanks !

**Comment by AlexandreAbraham** [comment]
The principle is that the splitter consider as lateralized a component with at least 95% of its voxels in one hémisphère. What you observés means that this région vas more than 5% of its voxel in the right hémisphère. I will check that tomorrow.

**Comment by GaelVaroquaux** [comment]
> The principle is that the splitter consider as lateralized a component
> with at least 95% of its voxels in one hémisphère.

I think that this is a good principle. Of course the 95% threshold is a
bit of a magic value that will never fall right for every situation.
Maybe it can be hand-tuned manually to have a good resulting atlas.

**Comment by salma1601** [comment]
Everything works great for cortical atlases. Seems to me it is just as you said @AlexandreAbraham
> Already splitted atlases should not be splitted again. The atlas you plot is actually not modified in our but the list of labels is modified even when it should not be.

**Comment by AlexandreAbraham** [comment]
Do you use the last version of the PR? I can't reproduce your problem.

File: nilearn/datasets/tests/test_atlas.py [hunk 1]
```diff
@@ -109,20 +109,44 @@ def test_fail_fetch_atlas_harvard_oxford():
...
     ho = atlas.fetch_atlas_harvard_oxford(target_atlas,
-                                          data_dir=tst.tmpdir)
-
-    assert_true(isinstance(nibabel.load(ho.maps), nibabel.Nifti1Image))
-    assert_true(isinstance(ho.labels, np.ndarray))
-    assert_true(len(ho.labels) > 0)
+                                          data_dir=tst.tmpdir,
+                                          symmetric_split=True)
+
+    assert_true(isinstance(ho.maps, nibabel.Nifti1Image))
```
Comment (banilo) [inline nilearn/datasets/tests/test_atlas.py]: Perhaps add check on ``ho.description``?

**Comment by salma1601** [comment]
>Do you use the last version of the PR?

No, sorry !! everything looks right now including for subcortical atalses

**Comment by AlexandreAbraham** [comment]
Merging then!

**Pull request merged**